packages/block-library/src/embed/transforms.js

```diff
@@ -33,11 +32,17 @@ const transforms = {
 		{
 			type: 'block',
 			blocks: [ 'core/paragraph' ],
+			isMatch: ( { url } ) => !! url,
 			transform: ( { url, caption } ) => {
-				const link = <a href={ url }>{ caption || url }</a>;
-				return createBlock( 'core/paragraph', {
-					content: renderToString( link ),
-				} );
+				const values = [ `<a href="${ url }">${ url }</a>` ];
+				if ( caption?.trim() ) {
+					values.push( caption );
+				}
+				return values.map( ( value ) =>
```

Comment:
Alternatively we could join the values with a `<br>` element so it remains a single block.  See https://github.com/WordPress/gutenberg/pull/32355#discussion_r649176795. I'll let you decide. :)

Reply:
Or even just a space. As long as the caption is not linked I guess. :)

Reply:
I updated to have a single block with a `<br>`.